pkg/workflow/workload/pod/builder.go

```diff
@@ -994,8 +1036,8 @@ func (m *Builder) ApplyResourceRequirements() error {
 		requirements = m.wf.Spec.Resources
 	}
 
-	m.pod.Spec.InitContainers = applyResourceRequirements(m.pod.Spec.InitContainers, requirements, false)
-	m.pod.Spec.Containers = applyResourceRequirements(m.pod.Spec.Containers, requirements, true)
+	m.pod.Spec.InitContainers = applyResourceRequirements(m.pod.Spec.InitContainers, requirements, false, config.Config.WorkloadCustomContainerCPUWeight)
```

Comment:
Init containers executes one by one, so I think it is not necessary to comply with this rule.